src/__tests__/index.test.ts

```diff
@@ -828,6 +828,13 @@ const vectors: Array<TestVector> = [
       { text: 'hs1qd42hrldu5yqee58se4uj6xctm7nk28r70e84vx', hex: '6d5571fdbca1019cd0f0cd792d1b0bdfa7651c7e' },
     ],
   },
+  {
+    name: 'STX',
+    coinType: 5757,
+    passingVectors: [
+      { text: 'SP2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKNRV9EJ7', hex: 'a46ff88886c2ef9762d970b4d2c63678835bd39d' },
```

Comment:
Can you put reference on where you got this test address? Their doc site has address starting `STJ` and their test explorer starts with `STB` https://docs.blockstack.org/understand-stacks/accounts https://testnet-explorer.blockstack.org/txid/0xa064d144f262f3f42af644c176bd4cc62554206112b4faa44353220274d63238

Reply:
Their explorer is for testnet, and the example address in their docs is for testnet(ST prefix) as well.
Versions for mainnet can be found here: https://github.com/blockstack/c32check/blob/master/src/address.ts#L6
Test case: https://github.com/blockstack/c32check/blob/master/tests/unitTests/src/index.ts#L479

The library I used to provide the versions are directly mentioned in the [docs](https://docs.blockstack.org/understand-stacks/accounts#creation:~:text=For%20Stacks%20addresses%2C%20c32check%20is%20used.%20Deriving).

A possible problem is that encoder/decoder is working only for p2pkh-mainnet and not for p2sh-mainnet(In their testnet explorer all addresses were p2pkh-testnet, so I went with p2pkh).
Since version&checksum is added after the actual encoding/decoding and has no difference except for the version provided to the function, I was unable to find a way to differentiate if the hex provided is p2pkh or p2sh.

For example:
-Version- -Hex payload- -Checksum-
SP 2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKK NRV9EJ7(p2pkh)
SM 2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKK QVX8X0G(p2sh)
Both gives the same hex when decoded:
a46ff88886c2ef9762d970b4d2c63678835bd39d
While encoding the hex, only difference is the version provided.

Even if we directly use their own library, the same problem occurs.

Reply:
@Echolon166 Bitcoin's canonical binary address encoding is a spend script, rather than just the decoded result of base58check; we decode and then transform into a script hash. Does STX have an equivalent to this?